### PR TITLE
Made each menu style emit open/close event. Add extra conditions for documentClick function. Modify handler for watcher for isOpen.

### PR DESCRIPTION
**Pull request by JakeFDev**
I am using this for a project and noticed that the open/close events are not being emitted for most of the menu variations.

I also fixed up some logic here and there.

### CODE REVIEW

File: src/components/Menu/bubble.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
     <div>
-        <Menu v-bind="propsToPass">
+        <Menu v-bind="propsToPass" openMenu="openMenu" @closeMenu="closeMenu">
```
Comment (JakeFDev) [inline src/components/Menu/bubble.vue]: Some of these components have the closeMenu and openMenu methods, but are not watching for the Menu to emit it. So it never calls.

File: src/components/Menu/push.vue [hunk 2]
```diff
@@ -21,6 +21,7 @@
             this.$emit("closeMenu")
         },
         push() {
+          this.openMenu()
```
Comment (JakeFDev) [inline src/components/Menu/push.vue]: Components with push/pull need to call openMenu and closeMenu in push/pull respectively

**Comment by JakeFDev** [comment]
@mbj36 This is my first PR so im not sure if I am able to add you as a reviewer (showing no reviewers)

**Comment by mbj36** [comment]
@JakeFDev I checked out this PR and everything looks good. Thanks for help.

**Pull request merged**